### PR TITLE
Fix issue with rich-text when nested in jsx objects

### PR DESCRIPTION
**Pull request by jeffsee55**
Closes https://github.com/tinacms/tinacms/issues/4484

### CODE REVIEW

File: packages/@tinacms/mdx/src/parse/acorn.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ const extractAttribute = (
         return extractString(attribute, field)
       }
     case 'object':
-      return extractObject(extractExpression(attribute), field)
+      return extractObject(extractExpression(attribute), field, imageCallback)
```
Comment (JackDevAU) [inline packages/@tinacms/mdx/src/parse/acorn.ts:81]: Should we track this (add a todo?) and make an issue to refactor this at a latter stage?

File: packages/@tinacms/mdx/src/parse/acorn.ts [hunk 2]
```diff
@@ -120,28 +120,30 @@ const extractScalar = <
 
 const extractObject = <T extends Extract<TinaField, { type: 'object' }>>(
   attribute: ExpressionStatement,
-  field: T
+  field: T,
+  imageCallback: (image: string) => string
```
Comment (JackDevAU) [inline packages/@tinacms/mdx/src/parse/acorn.ts:124]: Should we track this (add a todo?) and make an issue to refactor this at a latter stage?

**Review by JackDevAU — APPROVED** [review]
lgtm :)

File: .changeset/old-bananas-walk.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@tinacms/mdx": patch
+tinacms: patch
+---
+
+Fix issue where rich-text nested inside JSX objects wasn't being parsed/stringified properly
```
Comment (jeffsee55) [inline .changeset/old-bananas-walk.md:6]:
```suggestion
Fix issue where rich-text nested inside JSX objects wasn't being parsed/stringified properly.
```

**Review by JackDevAU — APPROVED** [review]

**Pull request merged**